src/client/tile.cpp

```diff
@@ -1802,6 +1805,157 @@ bool TextureSource::generateImagePart(std::string part_of_name,
+
+			core::rect<s32> oldViewPort = driver->getViewPort();
+			core::matrix4 oldProjMat = driver->getTransform(video::ETS_PROJECTION);
+			core::matrix4 oldViewMat = driver->getTransform(video::ETS_VIEW);
+			core::matrix4 oldWorldMat = driver->getTransform(video::ETS_WORLD);
+
+			video::ITexture* rt = driver->addRenderTargetTexture(img_dim, "render_invimg", video::ECF_A8R8G8B8);
+			if (!rt) {
+				errorstream << "generateImagePart(): Generating rendertarget failed "
+						<< "for part_of_name=\"" << part_of_name
+						<< "\", cancelling." << std::endl;
+				return false;
+			}
+			driver->setRenderTarget(rt, true, true, video::SColor(0,0,0,0));
+
+			scene::IMesh *mesh = imesh->mesh;
+			driver->clearZBuffer();
+
+			core::matrix4 ProjMatrix;
+			ProjMatrix.buildProjectionMatrixOrthoLH(2.0f, 2.0f, -1.0f, 100.0f);
+
+			core::matrix4 ViewMatrix;
+			ViewMatrix.buildProjectionMatrixOrthoLH(2.0f, 2.0f, -1.0f, 100.0f);
+			ViewMatrix.setTranslation(core::vector3df(
+				1.0f * (res_rect.LowerRightCorner.X + res_rect.UpperLeftCorner.X -
+						res_rect.LowerRightCorner.X - res_rect.UpperLeftCorner.X) /
+						res_rect.getWidth(),
+				1.0f * (res_rect.LowerRightCorner.Y + res_rect.UpperLeftCorner.Y -
+						res_rect.LowerRightCorner.Y - res_rect.UpperLeftCorner.Y) /
+						res_rect.getHeight(),
+				0.0f));
+
+			driver->setTransform(video::ETS_PROJECTION, ProjMatrix);
+			driver->setTransform(video::ETS_VIEW, ViewMatrix);
+
+			core::matrix4 matrix;
+			matrix.makeIdentity();
+
+			driver->setTransform(video::ETS_WORLD, matrix);
+			driver->setViewPort(res_rect);
+
+			video::SColor basecolor = def.color;
+
+			u32 mc = mesh->getMeshBufferCount();
+			for (u32 j = 0; j < mc; ++j) {
+				scene::IMeshBuffer *buf = mesh->getMeshBuffer(j);
+				// we can modify vertices relatively fast,
+				// because these meshes are not buffered.
+				assert(buf->getHardwareMappingHint_Vertex() == scene::EHM_NEVER);
+				video::SColor c = basecolor;
+
+				if (imesh->buffer_colors.size() > j) {
+					ItemPartColor *p = &imesh->buffer_colors[j];
+					if (p->override_base)
+						c = p->color;
+				}
+
+				if (imesh->needs_shading)
+					colorizeMeshBuffer(buf, &c);
+				else
+					setMeshBufferColor(buf, c);
+
+				video::SMaterial &material = buf->getMaterial();
+				material.MaterialType = video::EMT_TRANSPARENT_ALPHA_CHANNEL_REF;
+				material.Lighting = false;
+				driver->setMaterial(material);
+				driver->drawMeshBuffer(buf);
+			}
+
+			// draw the inventory_overlay
+			if (def.type == ITEM_NODE && def.inventory_image.empty() &&
+					!def.inventory_overlay.empty()) {
+				video::ITexture *overlay_texture = getTexture(def.inventory_overlay);
+				core::dimension2d<u32> dimens = overlay_texture->getOriginalSize();
+				core::rect<s32> srcrect(0, 0, dimens.Width, dimens.Height);
+				draw2DImageFilterScaled(driver, overlay_texture, res_rect, srcrect, &res_rect, 0, true);
+			}
+
+			driver->setRenderTarget(0, false, false, 0);
```

Comment:
Compile error, ambiguous function

Reply:
I'm getting no compile error, but your suggestion complies better with the coding style, so I will commit it.